app/controllers/contractors_controller.rb

```diff
@@ -18,8 +18,13 @@ def create
   end
 
   def show
-    render file: 'public/404.html' unless current_contractor
-    @contractor = current_contractor
+    if params[:contractor_id]
+      @contractor = Contractor.find(params[:contractor_id])
+      render :public
+    else
+      render file: 'public/404.html' unless current_contractor
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.